src/login/logind-button.c

```diff
@@ -395,12 +505,21 @@ static int button_suitable(int fd) {
                     bitset_get(keys, KEY_POWER2) ||
                     bitset_get(keys, KEY_SLEEP) ||
                     bitset_get(keys, KEY_SUSPEND) ||
-                    bitset_get(keys, KEY_RESTART))
+                    bitset_get(keys, KEY_RESTART) ||
+                    bitset_get(keys, KEY_LEFTSHIFT) ||
+                    bitset_get(keys, KEY_RIGHTSHIFT) ||
+                    bitset_get(keys, KEY_LEFTCTRL) ||
+                    bitset_get(keys, KEY_RIGHTCTRL) ||
+                    bitset_get(keys, KEY_LEFTMETA) ||
+                    bitset_get(keys, KEY_RIGHTMETA) ||
+                    bitset_get(keys, KEY_LEFTALT) ||
+                    bitset_get(keys, KEY_RIGHTALT) ||
+                    bitset_get(keys, KEY_ESC))
```

Comment:
given this is now growing to ridiculous sizes, please maintain a static const array somwhere with all the keys we care about, and then iterate through that array here.